Rubberduck.UnitTesting/UnitTesting/TestEngine.cs

```diff
@@ -146,7 +131,10 @@ private void RunInternal(IEnumerable<TestMethod> tests)
             }
             // FIXME we shouldn't need to handle awaiting a certain parser state ourselves anymore, right?
             // that would drop the _testsRequested member completely
+            // _testsRunning avoids raising a TestsRefreshed event when exiting the parser suspension
+            _testsRunning = true;
             _state.OnSuspendParser(this, AllowedRunStates, () => RunWhileSuspended(tests));
+            _testsRunning = false;
```

Comment:
Shouldn't the `_testsRunning` be at the end of the `RunWhileSuspended(tests`) action? The suspension action will run async and thus return immediately?

Reply:
Consider that I'm the bozo who wrote it (but only after having gotten extensive help), I think I shot my own feet since the `On....` idiom strongly implied it would run as an event. Disregard my earlier comment.

Even if it worked, it sounds very counterintuitive to include `ParserState.Suspended` in the `AllowedRunStates`, likely violating the principle of least astonishment. Also, keep in mind that you have no guarantee of what state you will enter after you exit the `ParserState.Suspended`. It might be the same state you had before you entered but that's not guaranteed, AFIUI, particularly if the suspension action was enter in from a different state other than `ParserState.Ready`.

Reply:
Yea it's a bit of a design problem that the UnitTesting can start running on states that force a full reparse when exiting the suspension... That results in the somewhat unintuitive behaviour of throwing away the results with the next parse. Said "next parse" is triggered directly when the unit-testing finishes...